source/common/http/http1/codec_impl.cc

```diff
@@ -436,12 +435,16 @@ void ConnectionImpl::completeLastHeader() {
     sendProtocolError(Http1ResponseCodeDetails::get().TooManyHeaders);
     const absl::string_view header_type =
         processing_trailers_ ? Http1HeaderTypes::get().Trailers : Http1HeaderTypes::get().Headers;
-    throw CodecProtocolException(absl::StrCat(header_type, " size exceeds limit"));
+    codec_exception_ = absl::InternalError(absl::StrCat(header_type, " size exceeds limit"));
+    // Crash if not dispatching_, otherwise exit early from the  parser.
+    RELEASE_ASSERT(dispatching_, "CodecProtocolException occurred outside of dispatching");
```

Comment:
If we're going to do this I'd prefer we have ASSERTS under completeLastHeader() and all friends that they're below dispatching as well.
Basically let's have fuzzing give us defense in depth for functions which should only be called here.

Reply:
I know that this will likely break something, and I'm a little confused by our model of throwing exceptions -- `RequestEncoderImpl::encodeHeaders` gets called outside the scope of `dispatching_` pretty often, and throws when there's missing path/method headers. I'm guessing that when it's called outside of `dispatching_`, we already have the guarantee that these exist?

Reply:
If so, maybe I should put ASSERT's in the branches where we throw

Reply:
>  know that this will likely break something, and I'm a little confused by our model of throwing exceptions -- RequestEncoderImpl::encodeHeaders gets called outside the scope of dispatching_ pretty often, and throws when there's missing path/method headers.

TBH I don't recall the history here. That "exception" shouldn't be an exception, it should just be an ASSERT/crash/etc. because it should never happen, so feel free to fix that however you want.

Reply:
You're right, it's only the encode pathways that may break something. I added ASSERTs!